extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/serde/tableExtractors.scala

```diff
@@ -240,3 +242,38 @@ class TableTableExtractor extends TableExtractor {
     lookupExtractor[StringTableExtractor].apply(spark, tableName)
   }
 }
+
+class HudiDataSourceV2RelationTableExtractor extends TableExtractor {
+  override def apply(spark: SparkSession, v1: AnyRef): Option[Table] = {
+    val outputObjs = new ArrayBuffer[PrivilegeObject]
+    PrivilegesBuilder.buildQuery(invokeAs[LogicalPlan](v1, "table"), outputObjs, spark = spark)
```

Comment:
What's the plan here and why we have to use `PrivilegesBuilder.buildQuery` here inside calling the `PrivilegesBuilder` ? 
Sounds bad with possible recursion and it's bad for decoupling.

Reply:
> What's the plan here and why we have to use `PrivilegesBuilder.buildQuery` here inside calling the `PrivilegesBuilder` ? Sounds bad with possible recursion and it's bad for decoupling.

Since here the sourceTable can be a query plan, so use `PrivilegeBuilder.buildQuery` is suitable...

Reply:
Changed